### ISSUE TITLE
Records with no properties

### ISSUE THREAD

**Issue by PetrSklenicka** [openrewrite/rewrite#3262]
Java record with no arguments is broken after I run my custom recipe.

Imagine I have the following record:

```
record MyRecord() {

}
```
It does not have any properties, but still it is perfectly valid piece of code. I implemented a custom `Recipe`, but even if I change my recipe to not doing anything, this record is modified. The result is following:

```
record MyRecord()) {

}
```
You can see there is an extra `)` added which makes the code uncompilable. It can be easily simulated with this test:

```
@Test
void simulateBug() {
    rewriteRun(
        java(
            """
                record MyRecord() {

                }
            """
        )
    );
}
```

The output:

```
org.opentest4j.AssertionFailedError: [When parsing and printing the source code back to text without modifications, the printed source didn't match the original source code. This means there is a bug in the parser implementation itself. Please open an issue to report this, providing a sample of the code that generated this error!] 
expected: 
  "record MyRecord() {
  
  }"
 but was: 
  "record MyRecord()) {
  
  }"
```

**Comment by knutwannheden** [comment]
@PetrSklenicka Thanks for reporting. This appears to be a duplicate of https://github.com/openrewrite/rewrite/issues/2956 which was fixed in [7.38.0](https://github.com/openrewrite/rewrite/releases/tag/v7.38.0). Can you verify that you are at least using this version? In case you are using the Maven plugin or the Gradle plugin I recommend that you upgrade to the latest version of those. See https://github.com/openrewrite/rewrite-maven-plugin/releases and https://github.com/openrewrite/rewrite-gradle-plugin/releases.

**Comment by PetrSklenicka** [comment]
Yes, thanks! Sorry for this.
You're right, in the 7.38.0 it is already fixed (I was using 7.37.3). 
Thanks.
